src/coreclr/src/jit/gentree.h

```diff
@@ -783,8 +783,21 @@ struct GenTree
 
 #define GTF_VAR_ARR_INDEX   0x00000020 // The variable is part of (the index portion of) an array index expression.
                                        // Shares a value with GTF_REVERSE_OPS, which is meaningless for local var.
-
-#define GTF_LIVENESS_MASK (GTF_VAR_DEF | GTF_VAR_USEASG | GTF_REG_BIRTH | GTF_VAR_DEATH)
+#define GTF_VAR_MULTIREG    0x20000000 // This is a struct or (on 32-bit platforms) long variable that is used or defined
+                                       // to/from a multireg source or destination (e.g. a call arg or return, or an op
+                                       // that returns its result in multiple registers such as a long multiply).
+// Last-use bits for a MULTIREG var. Note that a node marked GTF_VAR_MULTIREG can only be a pure
+// definition of all the fields, or a pure use of all the fields, so we don't need the equivalent of
+// GTF_VAR_USEASG.
+#define GTF_VAR_MULTIREG_DEATH1 0x00040000 // GT_LCL_VAR -- The last-use bit for the second register of a multireg lclVar.
+#define GTF_VAR_MULTIREG_DEATH2 0x00080000 // GT_LCL_VAR -- The last-use bit for the third register of a multireg lclVar.
+#define GTF_VAR_MULTIREG_DEATH3 0x00100000 // GT_LCL_VAR -- The last-use bit for the fourth register of a multireg lclVar.
+#define GTF_VAR_MULTIREG_DEATH_MASK (GTF_VAR_MULTIREG_DEATH1 | GTF_VAR_MULTIREG_DEATH2 | GTF_VAR_MULTIREG_DEATH3)
```

Comment:
Is the first register controlled with `GTF_VAR_DEATH`? Won't it be simpler to add `GTF_VAR_MULTIREG_DEATH0`  or change `GTF_VAR_DEATH` value to `0x00040000` to do `#define GTF_VAR_MULTIREG_DEATH0 GTF_VAR_DEATH`?

Reply:
Yes, the first register is controlled with `GTF_VAR_DEATH`. It might be better to change the bit used for that - I'd have to check what bits are available.